src/Core.php

```diff
@@ -324,6 +339,33 @@ public static function deactivate(): void {
 		return;
 	}
 
+	/**
+	 * @param string $host
+	 * @param array  $allowed_hosts
+	 */
+	private function is_allowed_host( string $host, array $allowed_hosts = [] ): bool {
+		if ( empty( $allowed_hosts ) ) {
+			$allowed_hosts = array_merge( YouTube::ALLOWED_HOSTS, Vimeo::ALLOWED_HOSTS, Dailymotion::ALLOWED_HOSTS );
+		}
+
+		/**
+		 * Allows to inject custom provider hosts
+		 * @var array $allowed_hosts List of allowed hosts
+		 * @var string $host         Current video hostname
+		 */
+		$allowed_hosts = apply_filters( 'tribe-embeds_allowed_provider_hosts', $allowed_hosts, $host );
+
+		if ( in_array( $host, $allowed_hosts ) ) {
+			return true;
+		}
+
+		$allowed_hosts = array_filter( $allowed_hosts, static function ( $allowed ) use ( $host ) {
+			return (bool) preg_match( "/$allowed/i", $host );
+		} );
```

Comment:
I think I haven't had enough coffee yet. Help me understand this filter. :)

Maybe add a comment block here explaining it?